addons/resource/models/resource.py

```diff
@@ -987,6 +989,87 @@ def _get_unavailable_intervals(self, start, end):
             resource_mapping.update(resources_unavailable_intervals)
         return resource_mapping
 
+    def _leaves_interval_gantt_view(self, start_dt, end_dt, leaves, attendances):
```

Comment:
If this code is related to gantt view, I don't get why it is defined in a low-level community application.

Reply:
I did not find a better module to do so. Since this change affect project_enterprise and planning, i had to find a common module for both of them to avoid duplicating the code. The module resource seemed more appropriate than the web_gantt, which seems to be very generic too.